src/rantanplan/core.py

```diff
@@ -98,6 +99,71 @@ def have_prosodic_liaison(first_syllable, second_syllable):
             and second_syllable['syllable'][0] in LIAISON_SECOND_PART)
 
 
+def get_synalephas(words):
+    prosodic_line = []
+    word_list = copy.deepcopy(words)
+    for idx, syllables_list in enumerate(word_list):
+        for syllable in syllables_list:
+            if not syllable.get('has_synalepha'):
+                prosodic_line.append(syllable)
+            else:
+                next_syllable = word_list[idx + 1].pop(0)
+                prosodic_line.append({'is_stressed': (syllable.get('is_stressed') or next_syllable.get('is_stressed')),
+                                      'syllable': "".join([syllable.get('syllable'), next_syllable.get('syllable')]),
+                                      'has_synalepha': True})
+    return prosodic_line
+
+
+def get_sinaeresis(word):
+    sinaeresis_words = word[:]  # create a copy since we're going to pop()
+    prosodic_syllables = []
+    for idx, key in enumerate(sinaeresis_words):
+        if not key.get('has_sinaeresis'):
+            prosodic_syllables.append(sinaeresis_words[idx])
+        else:
+            next_syl = sinaeresis_words.pop(idx + 1)
+            prosodic_syllables.append({'is_stressed': (key['is_stressed'] or next_syl['is_stressed']),
+                                       'syllable': "".join([key['syllable'], next_syl['syllable']]),
+                                       'has_sinaeresis': True})
+    return prosodic_syllables
+
+
+def get_metre(poem):
+    for lines in poem:
+        metre = ""
+        token_list = lines['tokens']
+        sinaeresis_tokens = {'sinaeresis_tokens': []}
+        prosodic_tokens = {'prosodic_tokens': []}
+        for tokens in token_list:
+            if tokens.get('word'):
+                sinaeresis_tokens['sinaeresis_tokens'].append(get_sinaeresis(tokens.get('word')))
+        synalepha_token_list = get_synalephas(sinaeresis_tokens['sinaeresis_tokens'])
+        for tokens in synalepha_token_list:
+            if tokens.get('syllable'):
+                prosodic_tokens['prosodic_tokens'].append(tokens)
+        lines.update(prosodic_tokens)
+        for syllable in synalepha_token_list:
+            if syllable['is_stressed']:
+                metre += '+'
+            else:
+                metre += '-'
+        last_word = sinaeresis_tokens['sinaeresis_tokens'][-1]
+        print('last word: ', last_word)
+        last_syllable_stress = last_word[-1].get('is_stressed')
+        if last_syllable_stress:
+            metre += '-'
+        elif len(last_word) >= 3:  # Esdrújula
```

Comment:
Is there a term in English for `Esdrújula`? If so, add it as well, please.

Reply:
Yes, good old proparoxytone